### PR TITLE
Request Signature integration example

### PR DESCRIPTION
**Pull request by renatodelpupo**
Adds instructions and an example of integration with a server-side application for request signature purposes.

Was also added some code formatting and unit tests.

### CODE REVIEW

**Review by andresguedes — APPROVED** [review]

**Review by pedroabdenor — APPROVED** [review]
👍

File: react-web-app/src/App.js [hunk 1]
```diff
@@ -12,44 +13,59 @@ function App() {
...
-			if (!data.metadata) data.metadata = {};
-      data.metadata.urlScheme = `${params.get("urlScheme")}://`;
-      data.metadata.integrationContext = params.get("integrationContext");
+    if (params.get('integrationContext') && params.get('urlScheme')) {
+      if (!data.metadata) data.metadata = {};
+      data.metadata.urlScheme = `${params.get('urlScheme')}://`;
+      data.metadata.integrationContext = params.get('integrationContext');
     }
+
+    // sign request - instructions available in readme.md
```
Comment (lukevance) [inline react-web-app/src/App.js:49]: Perfect! 🙌

File: react-web-app/src/SelectBankCard.js [hunk 2]
```diff
@@ -1,23 +1,25 @@
...
 export default function SelectBankCard(props) {
+  useEffect(() => {
+    const { establishData, TrustlyOptions } = props;
 
-  React.useEffect(() => {
-    // const [establishData, TrustlyOptions] = props;
-    const data = props.establishData();
-    window.Trustly.selectBankWidget(data, props.TrustlyOptions);
-  });
+    (async () => {
```
Comment (lukevance) [inline react-web-app/src/SelectBankCard.js:10]: nice 😎

File: react-web-app/src/utils/signature.test.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import { getRequestSignature } from './signature';
+
+// This is just a mockup
+const isHmacSha1Base64Hash = (str) => {
+  if (!str) return;
+
+  // Regular expression for Base64 encoding
+  const isValidBase64 = (str) => /^[A-Za-z0-9+/]+={0,2}$/.test(str);
+
+  // Check if the length is 28 characters (without padding) or 40 characters (with padding)
```
Comment (lukevance) [inline react-web-app/src/utils/signature.test.js:10]: Are all request signatures exactly this length??

File: react-web-app/src/utils/signature.test.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+  return isLengthValid && isBase64Valid;
+};
+
+describe('getRequestSignature', () => {
+  test('returns a Base64 encoded HMAC-SHA1-like data if request succeeds', async () => {
+    global.fetch = jest.fn(() =>
+      Promise.resolve({
+        ok: true,
+        json: () => Promise.resolve('v0LmRqCUqgBTVV7fd7wWp0ZFLQQ='),
```
Comment (lukevance) [inline react-web-app/src/utils/signature.test.js:24]: What is this value?

**Review by lukevance — COMMENTED** [review]
Looks good! Just wondering about that one hardcoded value that looks like a signature, can you explain what that is? thanks!

File: react-web-app/src/utils/signature.test.js (hunk 3 above)
Comment (renatodelpupo) [reply react-web-app/src/utils/signature.test.js:10]: This method `isHmacSha1Base64Hash` is just a mockup/simulation example made for the unit test. It doesn't reflect the actual implementation as it is on the front end and doesn't depend on or is affected by the implementation adopted by the merchant on the back end.

File: react-web-app/src/utils/signature.test.js (hunk 4 above)
Comment (renatodelpupo) [reply react-web-app/src/utils/signature.test.js:24]: This value is a made-up result returned by the mocked `fetch()` inside this specific unit test. Is based on an example of a hash that meets the proposed doc requirements (Base64 encoded HMAC-SHA1).

Comment (lukevance) [reply react-web-app/src/utils/signature.test.js:24]: Of course 🤦 this makes perfect sense!

**Review by lukevance — APPROVED** [review]
LGTM!

**Pull request merged**